WalletWasabi.Tests/UnitTests/Userfacing/ParserTests.cs

```diff
@@ -185,33 +185,75 @@ public void BitcoinAddressParserTests()
 	[Fact]
 	public void BitcoinUrlParserTests()
 	{
-		(string url, Network network)[] tests = new[]
+		(string uri, Network network)[] invalid_uri_tests = new[]
 		{
-				("bitcoin:18cBEMRxXHqzWWCxZNtU91F5sbUNKhL5PX?amount=50&label=Luke-Jr&message=Donation%20for%20project%20xyz", Network.Main),
-				("bitcoin:17VZNX1SN5NtKa8UQFxwQbFeFc3iqRYhem?amount=50&label=Luke-Jr&message=Donation%20for%20project%20xyz", Network.Main),
-				("bitcoin:3EktnHQD7RiAE6uzMj2ZifT9YgRrkSgzQX?amount=50&label=Luke-Jr&message=Donation%20for%20project%20xyz", Network.Main),
-				("bitcoin:bc1qw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t4?amount=50&label=Luke-Jr&message=Donation%20for%20project%20xyz", Network.Main),
-				("bitcoin:mipcBbFg9gMiCh81Kj8tqqdgoZub1ZJRfn?amount=50&label=Luke-Jr&message=Donation%20for%20project%20xyz", Network.TestNet),
-				("bitcoin:2MzQwSSnBHWHqSAqtTVQ6v47XtaisrJa1Vc?amount=50&label=Luke-Jr&message=Donation%20for%20project%20xyz", Network.TestNet),
-				("bitcoin:tb1qw508d6qejxtdg4y5r3zarvary0c5xw7kxpjzsx?amount=50&label=Luke-Jr&message=Donation%20for%20project%20xyz", Network.TestNet),
-			};
-
-		foreach (var test in tests)
+			(null!, Network.Main),
+			("", Network.Main),
+			("nfdjksnfjkdsnfjkds", Network.Main),
+			("18cBEMRxXHqzWWCxZNtU91F5sbUNKhL5PX", Network.Main),
+			("18cBEMRxXHqzWWCxZNtU91F5sbUNKhL5PX?amount=20.3", Network.Main),
+			("bitcoin:", Network.Main),
+			("bitcoin:?amount=20.3", Network.Main),
+			("bitcoin:18cBEMRxXHqzWWCxZNtU91F5sbUNKhL5PX?amount=", Network.Main),
+			("bitcoin:18cBEMRxXHqzWWCxZNtU91F5sbUNKhL5PX?amount=XYZ", Network.Main),
+			("bitcoin:18cBEMRxXHqzWWCxZNtU91F5sbUNKhL5PX?amount=100'000", Network.Main),
+			("bitcoin:18cBEMRxXHqzWWCxZNtU91F5sbUNKhL5PX?amount=100,000", Network.Main),
+			("bitcoin:18cBEMRxXHqzWWCxZNtU91F5sbUNKhL5PX?req-somethingyoudontunderstand=50&req-somethingelseyoudontget=999", Network.Main),
+            // this one should fail as amount is above 21M, but currently it does not.
+			// ("bitcoin:18cBEMRxXHqzWWCxZNtU91F5sbUNKhL5PX?amount=100000000", Network.Main)
```

Comment:
If you try `92233720369` or above the test passes, so `92233720368` or lower is accepted as valid amount here.

`Int64.MaxValue` contains this `92233720368`.

https://learn.microsoft.com/en-us/dotnet/api/system.int64.maxvalue?view=net-7.0

`public const long MaxValue = 9223372036854775807;`

Reply:
But we should not accept amount values that exceed amount of bitcoins there will ever be in existence.

Reply:
I know that `92233720368` should not be accepted, I have just pointed out what I have found out while testing this, so it might help investigating this further.

Reply:
It's not so important maybe, there is check in frontend.

![image](https://user-images.githubusercontent.com/4500994/232089708-ebca96a3-52bf-45ec-ac9d-a2b241babaeb.png)